lib/absinthe/middleware/batch.ex

```diff
@@ -147,7 +147,19 @@ defmodule Absinthe.Middleware.Batch do
     end)
     |> Map.new(fn {batch_opts, task, start_time_mono, metadata} ->
       timeout = Keyword.get(batch_opts, :timeout, 5_000)
-      result = Task.await(task, timeout)
+
+      result =
+        try do
+          Task.await(task, timeout)
+        catch
+          :exit, {:timeout, {Task, :await, _}} ->
+            raise """
```

Comment:
I think the main thing I need to sort out is whether turning this exit into a raise is going to be an issue.

`Task.await/2` emits an exit when it times out, instead of a `RuntimeError` that `raise` emits.

@maartenvanvliet can you think of any reason why this could be a problem?